resources/fetchersManager/data.go

```diff
@@ -70,8 +72,12 @@ func (d *Data) fetchAndSleep(ctx context.Context) {
 		case <-ctx.Done():
 			d.log.Info("Fetchers manager canceled")
 			return
-		case <-time.After(d.interval):
-			d.fetchIteration(ctx)
+		case <-timer.C:
+			// upadte the interval
+			timer.Reset(d.interval)
+			// this is blocking so the stop will not be called until all the fetchers are finished
+			// in case there is a blocking fetcher it will halt (til the d.timeout)
+			go d.fetchIteration(ctx)
 		}
```

Comment:
I wonder if a graceful shutdown in which all the healthy fetchers completed their cycle is not better than stopping them immediately.
We can let the fetcher's timeout handle blocked fetchers, wdyt?

Reply:
You are right, graceful shutdown will break this way.